api/src/pcapi/core/offers/api.py

```diff
@@ -490,6 +490,19 @@ def update_offer(
     if offerer_address:
         fields["offererAddress"] = offerer_address
 
+    if "videoUrl" in fields:
+        if new_video_url := fields.pop("videoUrl", None):
+            videos_api.upsert_video_and_metadata(new_video_url, offer)
+        elif offer.metaData and offer.metaData.videoUrl:
+            videos_api.remove_video_data_from_offer_metadata(
+                offer.metaData, offer.id, offer.venueId, offer.metaData.videoUrl
+            )
+            logger.info(
+                "Video has been deleted from offer",
+                extra={"offer_id": offer.id, "venue_id": offer.venueId, "video_url": offer.metaData.videoUrl},
+                technical_message_id="offer.video.deleted",
+            )
+
```

Comment:
Il manque les API error ici, c'est voulu ? 

sinon je propose : 

```suggestion
if "videoUrl" in fields:
        if new_video_url := fields.pop("videoUrl", None):
            try:
                videos_api.upsert_video_and_metadata(new_video_url, offer)
            except videos_exceptions.InvalidVideoUrl:
                raise ApiErrors(
                    errors={
                        "videoUrl": [
                            "Veuillez renseigner une URL provenant de la plateforme Youtube. Les shorts et les chaînes ne sont pas acceptées."
                        ]
                    }
                )
            except videos_exceptions.YoutubeVideoNotFound:
                raise ApiErrors(
                    errors={"videoUrl": ["URL Youtube non trouvée, vérifiez si votre vidéo n’est pas en privé."]}
                )
        elif offer.metaData and offer.metaData.videoUrl:
            videos_api.remove_video_data_from_offer_metadata(
                offer.metaData, offer.id, offer.venueId, offer.metaData.videoUrl
            )
```

Reply:
Il me semblait qu'on voulait éviter de mettre des `ApiError` en dehors des contrôleurs (qui ne gèrent d'ailleurs peut-être pas ces erreurs).